### ISSUE TITLE
[a11y]: Pagination dropdown white-on-white for g90 and g100 themes on Windows

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#9625]
Refs https://github.com/carbon-design-system/carbon/issues/9112

### Package

carbon-components

### Browser

Chrome

### Operating System

Windows

### Package version

10.43

### Automated testing tool and ruleset

None, I don't think EAAC can check select dropdowns?

### Assistive technology

no

### Description

When you open one of Pagination's dropdowns (either items per page, or page number) and then **move the mouse away**, the dropdown becomes white-on-white:

<img width="700" alt="Screen Shot 2021-09-09 at 16 21 28" src="https://user-images.githubusercontent.com/69599/132641240-4688bf35-a624-4c55-82af-577ee588a52b.png">

NB: This only happens with a dark carbon theme (g90 or g100), **and** on Windows.

When hovering a `<select>` , Carbon sets the background color to `rgb(53, 53, 53)` and things work fine.

But when not hovering, Carbon sets each `<select>`'s `background-color` to `transparent`, which (non-intuitively) makes the dropdown gray on the Mac, but white on Windows.

This seems to fix it:

```
.#{$prefix}--select-option {
	background: $background-hover;
}
```

### WCAG 2.1 Violation

Minimum contrast

### CodeSandbox example

https://carbon-components-react.netlify.app/?path=/story/components-pagination--pagination

### Steps to reproduce

Access https://carbon-components-react.netlify.app/?path=/story/components-pagination--pagination and switch the carbon theme to g100.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Looks like this is currently an issue  with the `inline` variant of `Select` in Firefox as well 
<img width="803" alt="Screen Shot 2021-09-10 at 2 07 18 PM" src="https://user-images.githubusercontent.com/11928039/132851081-d1f82cbb-5fcb-4617-a6d2-7df57d29c0cb.png">

**Comment by wkeese** [comment]
Thanks!  I confirm that https://carbon-components-react.netlify.app/?path=/story/components-pagination--pagination is working, including g90 and g100.
